modelzoo/inception_v3.py

```diff
@@ -0,0 +1,200 @@
+# Inception-v3, model from the paper:
+# "Rethinking the Inception Architecture for Computer Vision"
+# http://arxiv.org/abs/1512.00567
+# Original source:
+# https://github.com/tensorflow/tensorflow/blob/master/tensorflow/models/image/imagenet/classify_image.py
+# License: http://www.apache.org/licenses/LICENSE-2.0
+
+# Download pretrained weights from:
+# https://s3.amazonaws.com/lasagne/recipes/pretrained/imagenet/inception_v3.pkl
+
+
+import theano.tensor as T
+
+from lasagne.layers import InputLayer
+from lasagne.layers import Conv2DLayer
+from lasagne.layers import Pool2DLayer
+from lasagne.layers import DenseLayer
+from lasagne.layers import GlobalPoolLayer
+from lasagne.layers import ConcatLayer
+from lasagne.layers import ElemwiseMergeLayer
+from lasagne.layers import ExpressionLayer
+from lasagne.layers.normalization import batch_norm
+from lasagne.nonlinearities import softmax
+
+
+def preprocess(im):
+    # Expected input: RGB uint8 image
+    # Input to network should be bc01, 299x299 pixels, scaled to [-1, 1].
+    import skimage.transform
+    import numpy as np
+
+    im = skimage.transform.resize(im, (299, 299), preserve_range=True)
+    im = (im - 128) / 128.
+    im = np.rollaxis(im, 2)[np.newaxis].astype('float32')
+
+    return im
+
+
+def bn_conv(input_layer, **kwargs):
+    l = Conv2DLayer(input_layer, **kwargs)
+    l = batch_norm(l, epsilon=0.001)
+    return l
+
+
+def avg_pool(input_layer, **kwargs):
+    # hack to work around https://github.com/Theano/Theano/issues/3776
```

Comment:
Has been shown to be a cuDNN v3 bug -- maybe use `average_exc_pad` now and live with the fact that it requires cuDNN v4? Theano gives an error if that mode is used with v3, so people will know.